core/src/main/c/share/dedup_comparers.h

```diff
@@ -52,12 +53,33 @@ class SortColumnComparer : dedup_column {
 public:
     inline int operator()(int64_t l, int64_t r) const {
         const T l_val = l > -1
-                           ? reinterpret_cast<T *>(column_data)[l]
-                           : reinterpret_cast<T *>(o3_data)[l & ~(1ull << 63)];
+                        ? reinterpret_cast<T *>(column_data)[l]
+                        : reinterpret_cast<T *>(o3_data)[l & ~(1ull << 63)];
 
         const T r_val = r > -1
-                           ? reinterpret_cast<T *>(column_data)[r]
-                           : reinterpret_cast<T *>(o3_data)[r & ~(1ull << 63)];
+                        ? reinterpret_cast<T *>(column_data)[r]
+                        : reinterpret_cast<T *>(o3_data)[r & ~(1ull << 63)];
+
+        // One of the values can be MIN of the type (null value)
+        // and subtraction can result in type overflow
+        return (l_val > r_val) - (l_val < r_val);
+    }
+};
+
+template<typename T, uint16_t SegmentBits, typename TIdx>
+class SortColumnComparerManyAddresses : dedup_column_many_addresses {
```

Comment:
can be a named template function or method of dedup_column. no benefits from operator()

Reply:
It's the approach you chose:
https://github.com/questdb/questdb/pull/3566#discussion_r1267812905